src/python/pants/base/worker_pool.py

```diff
@@ -103,7 +105,7 @@ def error(e):
 
     # We filter out Nones defensively. There shouldn't be any, but if a bug causes one,
     # Pants might hang indefinitely without this filtering.
-    work_iter = iter(filter(None, work_chain))
+    work_iter = iter([_f for _f in work_chain if _f])
```

Comment:
This could be just `work_iter = (_f for _f in work_chain if _f)`; ie directly construct an iter (generator) vs create a list and then ask it for an iter. If this change was automated however, I'm fine with it.

Reply:
I'll fix it. If we don't change it now, it will probably never get changed and might be confusing to later devs why we're using this roundabout approach.

I do agree with you that we should be more rigorous though with keeping 100% semantic parity (as possible). But for small things like removing extraneous `list()`, I think it's worth fixing what `futurize` generates.